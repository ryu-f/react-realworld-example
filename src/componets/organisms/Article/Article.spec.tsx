import * as React from 'react'

import { Article } from './'
import { MemoryRouter } from 'react-router'
import { render } from '@testing-library/react'

describe('Article', () => {
  const article = {
    title: 'test',
    slug: 'test-slug',
    body: 'test-body',
    createdAt: new Date('2020-03-21T05:56:25.787Z'),
    updatedAt: new Date('2020-03-21T05:56:25.787Z'),
    tagList: ['test1', 'test2'],
    description: 'test-description',
    author: {
      username: 'test-user',
      bio: 'test-bio',
      image: 'https://via.placeholder.com/1',
      following: false
    },
    favorited: false,
    favoritesCount: 0
  }
  test('snapshot', () => {
    const { asFragment } = render(
      <MemoryRouter>
        <Article article={article} />
      </MemoryRouter>
    )
    expect(asFragment()).toMatchSnapshot()
  })
})
