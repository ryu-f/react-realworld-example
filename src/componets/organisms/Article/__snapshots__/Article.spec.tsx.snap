// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Article 1 1`] = `
<DocumentFragment>
  <div
    class="sc-fzoLsD RwjOw"
  >
    <div
      class="sc-fzpans kLztrW"
    >
      <a
        class="sc-fzplWN kKHbyV"
        href="/user/test-user"
      >
        <img
          alt=""
          class="sc-AxhUy fRFEWe"
          loading="lazy"
          src="https://via.placeholder.com/1"
        />
      </a>
      <div
        class="sc-fznyAO jDRRPK"
      >
        <p
          class="sc-AxjAm sc-AxirZ cLbhoI sc-fznKkj gQEAuv"
          size="20"
        >
          test
        </p>
        <p
          class="sc-AxjAm sc-AxirZ hAlrWo sc-fznZeY cPNPgI"
          size="14"
        >
          test-description
        </p>
        <ul
          class="sc-fzqBZW bsWVik"
        >
          <li
            class="sc-fzqNJr kDyyJu"
          >
            <span
              class="sc-fzozJi diAmiS"
            >
              test1
            </span>
          </li>
          <li
            class="sc-fzqNJr kDyyJu"
          >
            <span
              class="sc-fzozJi diAmiS"
            >
              test2
            </span>
          </li>
        </ul>
        <div
          class="sc-fzokOt dqxLUb"
        >
          <p
            class="sc-AxjAm sc-AxirZ hAlrWo"
            size="14"
          >
            by 
          </p>
          <a
            class="sc-AxhCb bLNLny sc-fzoyAV ifDJKs"
            href="/user/test-user"
            size="16"
            textcolor="GLAY"
          >
            test-user
          </a>
          <p
            class="sc-AxjAm sc-AxirZ ikYLvZ"
            size="16"
          >
            ・Sat Mar 21
          </p>
        </div>
      </div>
    </div>
    <div
      class="sc-fzoLag bkLbgZ"
    >
      <p
        class="sc-AxjAm sc-AxirZ hAlrWo sc-fzoXzr bvUUDN"
        size="14"
      >
        1 min read
      </p>
      <a
        class="sc-AxmLO dauKBD sc-fzpjYC gbLMzn"
        href="/article/test-slug"
      >
        READ
      </a>
    </div>
  </div>
</DocumentFragment>
`;

exports[`Article snapshot 1`] = `
<DocumentFragment>
  <div
    class="sc-fzoLsD RwjOw"
  >
    <div
      class="sc-fzpans kLztrW"
    >
      <a
        class="sc-fzplWN kKHbyV"
        href="/user/test-user"
      >
        <img
          alt=""
          class="sc-AxhUy fRFEWe"
          loading="lazy"
          src="https://via.placeholder.com/1"
        />
      </a>
      <div
        class="sc-fznyAO jDRRPK"
      >
        <p
          class="sc-AxjAm sc-AxirZ cLbhoI sc-fznKkj gQEAuv"
          size="20"
        >
          test
        </p>
        <p
          class="sc-AxjAm sc-AxirZ hAlrWo sc-fznZeY cPNPgI"
          size="14"
        >
          test-description
        </p>
        <ul
          class="sc-fzqBZW bsWVik"
        >
          <li
            class="sc-fzqNJr kDyyJu"
          >
            <span
              class="sc-fzozJi diAmiS"
            >
              test1
            </span>
          </li>
          <li
            class="sc-fzqNJr kDyyJu"
          >
            <span
              class="sc-fzozJi diAmiS"
            >
              test2
            </span>
          </li>
        </ul>
        <div
          class="sc-fzokOt dqxLUb"
        >
          <p
            class="sc-AxjAm sc-AxirZ hAlrWo"
            size="14"
          >
            by 
          </p>
          <a
            class="sc-AxhCb bLNLny sc-fzoyAV ifDJKs"
            href="/user/test-user"
            size="16"
            textcolor="GLAY"
          >
            test-user
          </a>
          <p
            class="sc-AxjAm sc-AxirZ ikYLvZ"
            size="16"
          >
            ・Sat Mar 21
          </p>
        </div>
      </div>
    </div>
    <div
      class="sc-fzoLag bkLbgZ"
    >
      <p
        class="sc-AxjAm sc-AxirZ hAlrWo sc-fzoXzr bvUUDN"
        size="14"
      >
        1 min read
      </p>
      <a
        class="sc-AxmLO dauKBD sc-fzpjYC gbLMzn"
        href="/article/test-slug"
      >
        READ
      </a>
    </div>
  </div>
</DocumentFragment>
`;
